contrib/pmsnare/pmsnare.c

```diff
@@ -95,79 +97,97 @@ CODESTARTparse
 	assert(pMsg != NULL);
 	assert(pMsg->pszRawMsg != NULL);
 
-	/* check if this message is of the type we handle in this (very limited) parser
-
-	 find out if we have a space separated or tab separated for the first item
-	 if tab separated see if the second word is one of our expected tags
-	  if so replace the tabs with spaces so that hostname and syslog tag are going to be parsed properly
-	   optionally replace the hostname at the beginning of the message with one from later in the message
-	  else, wrong message, abort
-	 else, assume that we have a valid timestamp, move over to the syslog tag
-	  if that is tab separated from the rest of the message and one of our expected tags
-	   if so, replace the tab with a space so that it will be parsed properly
-	    optionally replace the hostname at the beginning of the message withone from later in the message 
-
-	*/
-	snaremessage=0;
-	lenMsg = pMsg->iLenRawMsg - pMsg->offAfterPRI;
+	/* Check if this message is of the type we handle in this parser. */
```

Comment:
I think it's useful to have the comments in place to explain what we are intending to do, especially in a case like this where the code can tell you what is happening, but not the why (the thinking behind the code)